packages/opal-client/opal_client/engine/runner.py

```diff
@@ -191,6 +217,18 @@ async def _run_process_until_terminated(self) -> int:
             )
         )
 
+        # After the process is up (PID exists) we also want to make sure the
+        # engine reports as healthy before we continue. We run the health
+        # check in the background and set an event so __aenter__ can await it.
+        async def _set_ready_when_healthy():
+            try:
+                await self._wait_for_engine_health()
+                self._engine_ready.set()
+            except Exception as e:
+                logger.error("Engine failed health check: {err}", err=e)
```

Comment:
doesn't an exception here mean that we need to kill the process and attempt again ?